django/db/models/options.py

```diff
@@ -825,3 +824,18 @@ def _get_fields(self, forward=True, reverse=True, include_parents=True, include_
         # Store result into cache for later access
         self._get_fields_cache[cache_key] = fields
         return fields
+
+    @property
+    def has_auto_field(self):
+        warnings.warn(
+            'Model._meta.has_auto_field is deprecated. Just check if the auto_field attribute is not None instead.',
+            RemovedInDjango20Warning, stacklevel=2
+        )
+        return self.auto_field is not None
+
+    @has_auto_field.setter
+    def has_auto_field(self, value):
+        warnings.warn(
+            'Model._meta.has_auto_field is deprecated. Just set the auto_field attribute only.',
```

Comment:
I'm not sure if there's much benefit to this warning. Removing it would allow third-party apps that set it to have warning free backwards-compatibility.